flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -782,35 +783,12 @@ private ResultFetcher callDropMaterializedTableOperation(
 
         if (CatalogMaterializedTable.RefreshStatus.ACTIVATED
                 == materializedTable.getRefreshStatus()) {
-            ContinuousRefreshHandler refreshHandler =
-                    deserializeContinuousHandler(materializedTable.getSerializedRefreshHandler());
-            // get job running status
-            JobStatus jobStatus = getJobStatus(operationExecutor, handle, refreshHandler);
-            if (!jobStatus.isTerminalState()) {
-                try {
-                    cancelJob(operationExecutor, handle, refreshHandler.getJobId());
-                } catch (Exception e) {
-                    jobStatus = getJobStatus(operationExecutor, handle, refreshHandler);
-                    if (!jobStatus.isTerminalState()) {
-                        throw new SqlExecutionException(
-                                String.format(
-                                        "Failed to drop the materialized table %s because the continuous refresh job %s could not be canceled."
-                                                + " The current status of the continuous refresh job is %s.",
-                                        tableIdentifier, refreshHandler.getJobId(), jobStatus),
-                                e);
-                    } else {
-                        LOG.warn(
-                                "An exception occurred while canceling the continuous refresh job {} for materialized table {},"
-                                        + " but since the job is in a terminal state, skip the cancel operation.",
-                                refreshHandler.getJobId(),
-                                tableIdentifier);
-                    }
-                }
+            if (CatalogMaterializedTable.RefreshMode.CONTINUOUS
+                    == materializedTable.getRefreshMode()) {
+                cancelContinuousRefreshJob(
+                        operationExecutor, handle, tableIdentifier, materializedTable);
             } else {
-                LOG.info(
-                        "No need to cancel the continuous refresh job {} for materialized table {} as it is not currently running.",
-                        refreshHandler.getJobId(),
-                        tableIdentifier);
+                deleteRefreshWorkflow(tableIdentifier, materializedTable);
             }
         } else if (CatalogMaterializedTable.RefreshStatus.INITIALIZING
```

Comment:
If the refresh status is `SUSPENDED`, we also need to drop the refresh workflow before drop the materialized table.